### ISSUE TITLE
Concern of efficacy of training code for content completion

### ISSUE THREAD

**Issue by dqj5182** [Kaihua-Chen/diffusion-vas#8]
The Diffusion-VAS model is trained to do content completion of occluded regions.
However, current training code doesn't fully train the model well in this context.

1. Synthetic occlusion patches are not imposed where the target object is already occluded in original image.
In your code, the synthetic occlusion patch is imposed only until IOU between modal mask and amodal mask is below 0.7. However, if you see this image, this already has IOU between modal mask and amodal mask below 0.7. This happens quite frequently in the SAIL-VOS dataset, resulting in not enough synthetic occlusion happening during training. Rather, it would be more reasonable to use IOU between "modal mask before synthetic occlusion" and "modal mask after synthetic occlusion" in order to appropriately do synthetic occlusion. I want to ask authors' opinion regarding this issue.

- RGB image
![Image](https://github.com/user-attachments/assets/b80efe30-e710-4147-a808-82c6a39c2b50)

- Amodal mask
![Image](https://github.com/user-attachments/assets/d54f81e9-36b1-4dbc-a04e-99d9ea852b1d)

- Modal mask
![Image](https://github.com/user-attachments/assets/06035395-e1fe-4814-90d0-83491bffe841)

2. The loss and training is not specifically targeting to do content completion of occluded image
With the "already occluded image" (but image that didn't have any synthetic occlusion patch yet), Diffusion-VAS is trained to generate the entire amodal mask region. But the problem is because many of the training samples contain "already occluded image", even when the training is done, it is learned to generate "already occluded image" rather than the image of the "actually occluded object". I am curious why the training is done in this manner although the objective of the paper is to do content completion of occluded object. May I ask why you chose this strategy instead of rather just training the model to generate visible modal regions (modal mask regions) before synthetic occlusion with the training pair of "image after synthetic occlusion" and "image before synthetic occlusion". Another option would be to strictly follow Figure 3 of this model's main paper which just uses samples that the target object doesn't have any occlusion and synthetic occlusion (if imposed) is the only occlusion present.

**Comment by Kaihua-Chen** [comment]
Thank you for pointing this out -- I made a mistake in the README. For content completion fine-tuning, `train_annot_path` and `eval_annot_path` should be same as `occluder_data_path`, i.e., `sailvos_complete_objs_as_occluders.json`. I’ve updated the README accordingly.

With this correction, the code strictly follows Section 3.4 and Figure 3 of our paper: when creating synthetic occlusions, we use only fully visible objects (visibility > 95%), where the modal and amodal masks are almost identical.

**Comment by dqj5182** [comment]
Thank you! It does seem to solve the problem I mentioned earlier. I will try to re-run the training with the fixed code and make another issue if there is any further problem.
